lib/resourcemerge/machineconfig.go

```diff
@@ -41,6 +41,18 @@ func EnsureMachineConfigPool(modified *bool, existing *mcfgv1.MachineConfigPool,
 		*modified = true
 		existing.Spec.MachineSelector = required.Spec.MachineSelector
 	}
+
+	if existing.Spec.MachineSelector != nil {
+		existing.Spec.NodeSelector = existing.Spec.MachineSelector
+	}
+	if existing.Spec.NodeSelector == nil {
```

Comment:
I also added these two `ifs` since the last push for NodeSelector because wouldn't we need it for new MCs?

Reply:
yes this is correct